src/deep-security/lib/CredentialsManager.js

```diff
@@ -170,16 +191,35 @@ export class CredentialsManager {
       configurable: true
     });
 
-    return JSON.stringify(credentials);
+    // save only credentials instead of all CognitoIdentityCredentials instance
+    let awsCredentials = {
+      expired: credentials.expired,
+      expireTime: credentials.expireTime,
+      accessKeyId: credentials.accessKeyId,
+      secretAccessKey: credentials.secretAccessKey,
+      sessionToken: credentials.sessionToken,
+    };
+
+    return JSON.stringify(awsCredentials);
   }
 
   /**
-   * @todo: implement an encoding method
+   * @todo: implement a decoding method
    *
    * @param {String} credentials
    * @returns {Object}
    */
   _decodeCredentials(credentials) {
-    return JSON.parse(credentials);
+    if (credentials && typeof credentials == 'string') {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/MitocGroup/deep-framework/file/1857611649/issues/source?bid=2895475&fileBranchId=3085948#l213)